tests/Collection/CrudSpecFunctionalTest.php

```diff
@@ -334,6 +370,50 @@ private function initializeData(array $initialData, array $expectedData = null)
         }
     }
 
+    private function prepareBulkWriteArguments($arguments)
+    {
+        $operations = [];
+        $operations['options'] = $arguments['options'];
+        foreach ($arguments['requests'] as $requests) {
+            if (isset($requests['arguments']['arrayFilters'])) {
+                if (isset($requests['arguments']['options'])) {
+                    array_push($requests['arguments']['options'], ['arrayFilters' => $requests['arguments']['arrayFilters']]);
+                } else {
+                    $options = ['arrayFilters' => $requests['arguments']['arrayFilters']];
```

Comment:
Looking at this some more, I think this is a bug. You're iterating through potentially many request elements, but this is always assigning an `$options` variable in the method's local scope. This means it's possible for one request's `arrayFilters` option to clobber `$options` from a previous request. Moreover, later in the method you have no way of knowing the exact request element to which the final value of `$options` relates.

It looks as if `$options` is only referenced from the `updateMany` and `updateOne` cases, so perhaps you just need to `unset()` it at the start of each iteration.